File: app/src/main/java/com/petnbu/petnbu/feed/FeedPhotosAdapter.java
package com.petnbu.petnbu.feed;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.google.android.gms.common.internal.Preconditions;
import com.petnbu.petnbu.BaseBindingViewHolder;
import com.petnbu.petnbu.GlideRequests;
import com.petnbu.petnbu.R;
import com.petnbu.petnbu.databinding.ViewFeedPhotosBinding;
import com.petnbu.petnbu.model.FeedUI;
import com.petnbu.petnbu.model.Photo;
import com.petnbu.petnbu.util.ImageUtils;

public class FeedPhotosAdapter extends RecyclerView.Adapter<FeedPhotosAdapter.PhotoHolder> {

    private FeedUI mFeed;
    private GlideRequests mGlideRequests;
    private PhotosPagerAdapter.OnItemClickListener mOnItemClickListener;
    private int mImageWidth;

    public FeedPhotosAdapter(FeedUI feed, GlideRequests glideRequests,
                             PhotosPagerAdapter.OnItemClickListener onItemClickListener, int imageWidth) {
        Preconditions.checkNotNull(feed);
        Preconditions.checkNotNull(glideRequests);

        mFeed = feed;
        mGlideRequests = glideRequests;
        mOnItemClickListener = onItemClickListener;
        mImageWidth = imageWidth;
    }

    @NonNull
    @Override
    public PhotoHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.view_feed_photos, parent, false);
        return new PhotoHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull PhotoHolder holder, int position) {
        holder.bindData(mFeed.getPhotos().get(position));
    }

    @Override
    public int getItemCount() {
        return mFeed.getPhotos() != null ? mFeed.getPhotos().size() : 0;
    }

    public void setData(FeedUI feed) {
        mFeed = feed;
        notifyDataSetChanged();
    }

    protected final class PhotoHolder extends BaseBindingViewHolder<ViewFeedPhotosBinding, Photo> {

        private PhotoHolder(View itemView) {
            super(itemView);
            ViewGroup.LayoutParams layoutParams = itemView.getLayoutParams();
            layoutParams.width = mImageWidth;
            itemView.setLayoutParams(layoutParams);
        }

        @Override
        public void bindData(Photo photo) {
            String photoUrl = ImageUtils.getPhotoUrl(photo, mImageWidth);
            mGlideRequests
                    .load(TextUtils.isEmpty(photoUrl) ? photo.getOriginUrl() : photoUrl)
                    .centerInside()
                    .into(mBinding.imgContent);
        }
    }
}

File: app/src/main/java/com/petnbu/petnbu/ui/feed/FeedPhotosAdapter.kt
package com.petnbu.petnbu.ui.feed

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.petnbu.petnbu.ui.BaseBindingViewHolder
import com.petnbu.petnbu.GlideRequests
import com.petnbu.petnbu.R
import com.petnbu.petnbu.databinding.ViewFeedPhotosBinding
import com.petnbu.petnbu.model.FeedUI
import com.petnbu.petnbu.model.Photo
import com.petnbu.petnbu.util.ImageUtils

class FeedPhotosAdapter(private var feed: FeedUI,
                        private val glideRequests: GlideRequests,
                        private val onItemClickListener: OnItemClickListener,
                        private val imageWidth: Int)
    : RecyclerView.Adapter<FeedPhotosAdapter.PhotoHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PhotoHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.view_feed_photos, parent, false)
        return PhotoHolder(view)
    }

    override fun onBindViewHolder(holder: PhotoHolder, position: Int) {
        holder.bindData(feed.getPhotos()[position])
    }

    override fun getItemCount() = feed.getPhotos()?.size ?: 0

    inner class PhotoHolder constructor(itemView: View) : BaseBindingViewHolder<ViewFeedPhotosBinding, Photo>(itemView) {

        init {
            itemView.layoutParams = itemView.layoutParams.apply {
                width = imageWidth
            }
        }

        override fun bindData(photo: Photo) {
            val photoUrl = ImageUtils.getPhotoUrl(photo, imageWidth)
            glideRequests
                    .load(if (!photoUrl.isNullOrBlank()) photoUrl else photo.originUrl)
                    .centerInside()
                    .into(mBinding.imgContent)
        }

        override fun bindData(item: Photo, payloads: List<Any>) {}
    }

    interface OnItemClickListener {

        fun onPhotoClicked()
    }
}
